src/dune/lib_config.mli

```diff
@@ -16,3 +16,5 @@ type t =
 val allowed_in_enabled_if : string list
 
 val get_for_enabled_if : t -> var:string -> string
+
+val emit_empty_archives : t -> bool
```

Comment:
The name is clear enough in the context of this PR, but I feel like it wouldn't immediately be obvious what it means to a new reader. What about `linker_can_create_empty_archives`?